app/src/main/database/index.ts

```diff
@@ -106,6 +106,33 @@ export class DB {
     this.deleteItem = this.db.prepare("DELETE FROM items WHERE uuid = @id");
   }
 
+  // Detect runtime environment
+  private isElectron(): boolean {
+    return (
+      typeof process !== "undefined" &&
+      process.versions !== undefined &&
+      process.versions.electron !== undefined
+    );
+  }
+
+  // Get binary path for current runtime
+  private getBinaryPath(): string {
+    const runtime = this.isElectron() ? "electron" : "node";
+    const basePath = path.join(
+      process.cwd(),
+      "node_modules",
+      "better-sqlite3",
+      "build",
+    );
+    const binaryPath = path.join(
+      basePath,
+      `Release-${runtime}`,
+      "better_sqlite3.node",
+    );
+    console.log(`Loading better-sqlite3 from ${binaryPath}`);
+    return binaryPath;
+  }
```

Comment:
This approach works for me across dev, tests, and package:

```suggestion
  private getBinaryPath(): string {
    const runtime = this.isElectron() ? "electron" : "node";
    const isPackaged = __dirname.includes("app.asar");

    const basePath = isPackaged
      ? path.join((process as any).resourcesPath, "app.asar.unpacked")
      : process.cwd();

    const binaryPath = path.join(
      basePath,
      "node_modules",
      "better-sqlite3",
      "build",
      `Release-${runtime}`,
      "better_sqlite3.node",
    );
    console.log(`Loading better-sqlite3 from ${binaryPath} (packaged: ${isPackaged})`);
    return binaryPath;
  }
```

Reply:
Thanks! Applied, just without the `(process as any)` which wasn't necessary.